File: SAFEVISION/safevision/safevision.tech/app/src/main/java/safevision/tech/SecondFragment.java
package safevision.tech;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;

import safevision.tech.databinding.FragmentSecondBinding;

public class SecondFragment extends Fragment {

    private FragmentSecondBinding binding;

    @Override
    public View onCreateView(
            @NonNull LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        binding = FragmentSecondBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setupNavigation();
    }

    private void setupNavigation() {
        binding.buttonSecond.setOnClickListener(v -> {
            try {
                NavHostFragment.findNavController(SecondFragment.this)
                        .navigate(R.id.action_SecondFragment_to_FirstFragment);
            } catch (Exception e) {
                e.printStackTrace();
                // Correction de la chaîne non fermée
                Toast.makeText(getContext(), "Erreur survenue lors de la navigation", Toast.LENGTH_SHORT).show();
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null; // Libération de la liaison pour éviter les fuites mémoire
    }
}

File: SAFEVISION/safevision/safevision.tech/app/src/main/java/safevision/tech/SecondFragment.kt
package safevision.tech

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import safevision.tech.databinding.FragmentSecondBinding

class SecondFragment : Fragment() {

    private var _binding: FragmentSecondBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Initialisation de la liaison de vues
        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Configuration des actions de navigation
        setupNavigation()
    }

    /**
     * Configure le bouton pour naviguer vers le fragment précédent.
     */
    private fun setupNavigation() {
        binding.buttonPrevious.setOnClickListener {
            navigateTo(R.id.action_secondFragment_to_firstFragment)
        }
    }

    /**
     * Navigue vers un autre fragment en utilisant l'identifiant de l'action.
     * @param actionId L'identifiant de l'action de navigation.
     */
    private fun navigateTo(actionId: Int) {
        try {
            findNavController().navigate(actionId)
        } catch (e: Exception) {
            e.printStackTrace()
            showToast("Erreur de navigation : ${e.message ?: "Action inconnue"}")
        }
    }

    /**
     * Affiche un message Toast.
     * @param message Le message à afficher.
     */
    private fun showToast(message: String) {
        Toast.makeText(requireContext(), message, Toast.LENGTH_SHORT).show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        // Nettoyage de la liaison de vues pour éviter les fuites de mémoire
        _binding = null
    }
}
